aws/rust-runtime/aws-config/src/profile/retry_config.rs

```diff
@@ -90,48 +92,67 @@ impl ProfileFileRetryConfigProvider {
+                return Ok(None);
+            }
+        };
+
         let selected_profile = self
             .profile_override
             .as_deref()
             .unwrap_or_else(|| profile.selected_profile());
-        let selected_profile = profile.get_profile(selected_profile)?;
+        let selected_profile = match profile.get_profile(selected_profile) {
+            Some(profile) => profile,
+            None => {
+                tracing::warn!("failed to get selected '{}' profile", selected_profile);
+                return Ok(None);
+            }
+        };
 
         let max_attempts = selected_profile
             .get("max_attempts")
             .and_then(|max_attempts| max_attempts.parse::<u32>().ok());
         let retry_mode = selected_profile.get("retry_mode");
 
-        match (retry_mode, max_attempts) {
-            // If neither are set, we can return None and move on
-            (None, None) => None,
-            // If at least one is set, we need to generate a `RetryConfig` and update it with the set value(s)
-            (retry_mode, max_attempts) => {
-                let mut retry_config = RetryConfig::new();
+        // If neither profile vars are set, we're done with this provider
+        if let (None, None) = (&max_attempts, retry_mode) {
+            return Ok(None);
+        }
 
-                if let Some(max_attempts) = max_attempts {
-                    assert_ne!(max_attempts, 0, "It is invalid to set max_attempts to 0. Unset it or set it to an integer greater than or equal to one.");
+        // If at least one is set, we need to generate a `RetryConfig` and update it with the set value(s)
+        let mut retry_config = RetryConfig::new();
 
-                    retry_config = retry_config.with_max_attempts(max_attempts);
-                }
+        if let Some(max_attempts) = max_attempts {
+            if max_attempts == 0 {
+                return Err(RetryConfigErr::MaxAttemptsMustNotBeZero {
+                    set_by: "aws profile".to_owned(),
+                });
+            };
 
-                if let Some(retry_mode) = retry_mode {
-                    match RetryMode::from_str(retry_mode).ok() {
-                        Some(retry_mode) => {
-                            assert_ne!(retry_mode, RetryMode::Adaptive, r#"Setting retry_mode to "adaptive" is not yet supported. Unset it or set it to a supported mode."#);
+            retry_config = retry_config.with_max_attempts(max_attempts);
+        }
 
-                            retry_config = retry_config.with_retry_mode(retry_mode);
-                        },
-                        None => panic!("It is invalid to set retry_mode to {}. Unset it or set it to a supported mode.", retry_mode),
-                    }
+        if let Some(retry_mode) = retry_mode {
+            match RetryMode::from_str(retry_mode) {
+                Ok(retry_mode) if retry_mode == RetryMode::Adaptive => {
```

Comment:
probably should just allowlist `Standard` for now since that's the only supported mode